sysfs/system_cpu.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sysfs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+// SystemCpuCpufreq contains stats from devices/system/cpu/cpu[0-9]*/cpufreq/...
+type SystemCpuCpufreq struct {
+	CurrentFrequency   uint64
+	MinimumFrequency   uint64
+	MaximumFrequency   uint64
+	TransitionLatency  uint64
+	AvailableGovernors string
+	Driver             string
+	Govenor            string
+	RelatedCpus        string
+	SetSpeed           string
+}
+
+// SystemCpufreq is a collection of SystemCpuCpufreq for every CPU.
+type SystemCpufreq map[string]SystemCpuCpufreq
+
+// TODO: Add topology support.
+
+// TODO: Add thermal_throttle support.
+
+// NewSystemCpufreq returns CPU frequency metrics for all CPUs.
+func NewSystemCpufreq() (SystemCpufreq, error) {
+	fs, err := NewFS(DefaultMountPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return fs.NewSystemCpufreq()
+}
+
+// NewSystemCpufreq returns CPU frequency metrics for all CPUs.
+func (fs FS) NewSystemCpufreq() (SystemCpufreq, error) {
```

Comment:
TBH I think my preference would be to return a slice with the CPU's name as a member of the structure.  Maps are usually a little gross in APIs.

Reply:
My personal preference would be a slice with field Name in both cases.  Maps can be clunky to use and the undefined map iteration order behavior bites a lot of people on a regular basis.